internal/kinds/capp-revision/adapters/capp_revision_adapter.go

```diff
@@ -0,0 +1,109 @@
+package adapters
+
+import (
+	"context"
+	"fmt"
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"math/rand"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strconv"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CappNameLabelKey   = "dana.rcs.io/cappName"
+	ClientListLimit    = 100
+	charSet            = "abcdefghijklmnopqrstuvwxyz0123456789"
+	RandomStringLength = 5
+	IndexCut           = 50
+)
+
+var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+// generateRandomString returns a random string of the specified length using characters from the charset.
+func generateRandomString(length int) string {
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charSet[seededRand.Intn(len(charSet))]
+	}
+	return string(b)
+}
+
+// getSubstringUntilIndex returns a substring from the start of the given string up to (but not including) the character at the specified index.
+// If the index is out of bounds, it returns the entire string.
+func getSubstringUntilIndex(s string, index int) string {
+	if index < 0 || index > len(s) {
+		return s
+	}
+	return s[:index]
+}
+
+// GetCappRevisions retrieves a list of CappRevision resources filtered by labels matching a specific Capp, returning the list and any error encountered.
+func GetCappRevisions(ctx context.Context, r client.Client, capp cappv1alpha1.Capp) ([]cappv1alpha1.CappRevision, error) {
+
+	cappRevisions := cappv1alpha1.CappRevisionList{}
+
+	requirement, err := labels.NewRequirement(CappNameLabelKey, selection.Equals, []string{capp.Name})
+	if err != nil {
+		return cappRevisions.Items, err
+	}
+
+	labelSelector := labels.NewSelector().Add(*requirement)
+	listOptions := client.ListOptions{
+		Namespace:     capp.Namespace,
+		LabelSelector: labelSelector,
+		Limit:         ClientListLimit,
+	}
+
+	err = r.List(ctx, &cappRevisions, &listOptions)
+	return cappRevisions.Items, err
+}
+
+// CreateCappRevision initializes and creates a CappRevision in Kubernetes from a Capp object, returning an error on failure.
```

Comment:
The "in Kubernetes" here is implied from the context of the project and is not needed here or in other places